flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/api/TableConfigOptions.java

```diff
@@ -179,81 +164,70 @@
+	public static final ConfigOption<String> SQL_EXEC_MINIBATCH_ALLOW_LATENCY =
+			key("sql.exec.mini-batch.allow-latency")
+					.defaultValue("-1 ms")
+					.withDescription("The maximum latency can be used for MiniBatch to buffer input records. " +
+							"MiniBatch is an optimization to buffer input records to reduce state access. " +
+							"MiniBatch is triggered with the allowed latency interval and when the maximum number of buffered records reached. " +
+							"If the allow-latency is greater than zero, MiniBatch will be enabled, otherwise, MiniBatch is disabled. " +
+							"NOTE: MiniBatch only works for non-windowed aggregations currently.");
 
 	public static final ConfigOption<Long> SQL_EXEC_MINIBATCH_SIZE =
 			key("sql.exec.mini-batch.size")
-					.defaultValue(Long.MIN_VALUE)
-					.withDescription("The maximum number of inputs that MiniBatch buffer can accommodate.");
-
-	// ------------------------------------------------------------------------
-	//  STATE BACKEND Options
-	// ------------------------------------------------------------------------
-
-	public static final ConfigOption<Boolean> SQL_EXEC_STATE_BACKEND_ON_HEAP =
-			key("sql.exec.statebackend.onheap")
-					.defaultValue(false)
-					.withDescription("Whether the statebackend is on heap.");
+					.defaultValue(-1L)
+					.withDescription("The maximum number of input records can be buffered for MiniBatch. " +
+							"MiniBatch is an optimization to buffer input records to reduce state access. " +
+							"MiniBatch is triggered with the allowed latency interval and when the maximum number of buffered records reached. " +
+							"NOTE: MiniBatch only works for non-windowed aggregations currently. Its value must be positive.");
 
 	// ------------------------------------------------------------------------
 	//  State Options
 	// ------------------------------------------------------------------------
-
-	public static final ConfigOption<Long> SQL_EXEC_STATE_TTL_MS =
-			key("sql.exec.state.ttl.ms")
-					.defaultValue(Long.MIN_VALUE)
+	public static final ConfigOption<String> SQL_EXEC_STATE_TTL =
+			key("sql.exec.state.ttl")
+					.defaultValue("-1 ms")
 					.withDescription("The minimum time until state that was not updated will be retained. State" +
 							" might be cleared and removed if it was not updated for the defined period of time.");
 
-	public static final ConfigOption<Long> SQL_EXEC_STATE_TTL_MAX_MS =
-			key("sql.exec.state.ttl.max.ms")
-					.defaultValue(Long.MIN_VALUE)
-					.withDescription("The maximum time until state which was not updated will be retained." +
-							"State will be cleared and removed if it was not updated for the defined " +
-							"period of time.");
-
 	// ------------------------------------------------------------------------
 	//  Other Exec Options
 	// ------------------------------------------------------------------------
-
+	@Documentation.ExcludeFromDocumentation(value = "Only for tests.")
```

Comment:
As far as I know. It is used in production widely.. User can force to use `SortMergeJoin`.